crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -373,3 +347,70 @@ impl Format<PyFormatContext<'_>> for FormatEmptyLines {
         }
     }
 }
+
+/// A helper used to measure formatted comments.
+///
+/// Use a temporary formatter to write a normalized, formatted comment
+/// to in order to compute its width for a reserved-width line suffix element.
+fn measure_comment(comment: &SourceComment, context: &PyFormatContext) -> FormatResult<u32> {
+    let mut state = FormatState::new(context.clone());
+    let mut buffer = VecBuffer::new(&mut state);
+    let comment_len = write_comment(&mut Formatter::new(&mut buffer), comment)?;
+    Ok(comment_len)
+}
```

Comment:
The measure comment now formats the comment and then throws it away. This means, every line suffix comment gets formatted twice. We could make some this faster by having a `DropAllBuffer` but formatting the comments twice still seems wasteful. 

That's why I think that `normalize_comments` works better. It has the disadvantage that it needs to allocate a `String` if the comment isn't already formatted (which wasn't the case before), but I think this is neglectable because most comments don't have extra leading whitespace.

Reply:
Do you have any feedback on a4d21fdcdc80d0409ee5931ea753ae9e2e603886?